Fix gtk extensions dir + install libnyxt.so + include source

Ambrevar opened the pull request:
3 commits for 3 related fixes:

- This fixes the GTK extension directory (was mistakenly made context-dependent in #2050) and sets it to something system-wide.  Indeed, we have only one such library, there is no point (for now at least) in supporting multiple libs.
- Install the lib together with the source code to a system-wide dir.  This partially addresses #2065.
- Update `nyxt-source-registry` to look into the newly installed source.

Thoughts?

This raises some questions, such as:

- Relying on `asdf-user::*data-dir*` is brittle.  Anything better?
- Should we introduce a new environment variable so that packagers have access to a more explicit configuration knob?

Ambrevar commented on nyxt.asd:
```diff
@@ -473,7 +475,28 @@ See `asdf::*immutable-systems*'."
-                (uiop:run-program (list "chmod" "+x" binary-file))))))
+                (uiop:run-program (list "chmod" "+x" binary-file)))
+              (let ((source-dir (uiop:ensure-directory-pathname (uiop:strcat *datadir* "/nyxt"))))
+                ;; TODO: Don't copy .fasls and .o?  Copy only the output of `git ls-files`, + libnyxt.so?
```
Thoughts on this?

aartaka replied:
Yes, git-visible files are likely the ones we need!

Ambrevar replied:
That said, `libnyxt.so` is not listed by Git.

Ambrevar commented:
Tests welcome!

aartaka commented on source/configuration.lisp:
```diff
@@ -56,6 +69,7 @@ This is set globally so that extensions can be loaded even if there is no
 (defun nyxt-source-registry ()
   `(:source-registry
     (:tree ,(nfiles:expand *extensions-directory*))
+    (:tree ,(nfiles:expand *source-directory*))
```
Ha, so it was that easy 0_o

Ambrevar commented:
Update:

- I've introduced 2 variables, one `*dest-source-dir*` in the .asd, another one `NYXT_SOURCE_PATH` (an environment variable).
- Added `git ls-files` support, falling back to copying the directories on failure.

Tested, seems to work perfectly fine.

Ambrevar commented:
I've tested by setting

```lisp
(setf (uiop:getenv "DESTDIR") "/tmp/nyxt-install/")
```

before loading the system.

Ambrevar commented:
OK to merge?

aartaka commented:
Had no chance to check :cold_sweat:

aartaka commented:
But well, if you say it works, I'll take your word for it :D
